src/Filter.Nest/QueryContainerDescriptorExtensions.cs

```diff
@@ -0,0 +1,99 @@
+﻿using Nest;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace RimDev.Filter.Nest
+{
+    public static class QueryContainerDescriptorExtensions
+    {
+        public static QueryContainer Filter<T>(
+            this QueryContainerDescriptor<T> value,
+            object filter,
+            IDictionary<Type, Func<object, string>> filterValueFormatters = null)
+            where T : class
+        {
+            if (filter == null)
```

Comment:
This majority of the method body is the same as [SearchDescriptorExtensions](https://github.com/ritterim/filter/blob/497cf748dd52d4d00631d550ea1ba35f56ee8585/src/Filter.Nest/SearchDescriptorExtensions.cs#L24-L103). If we expect both types of `Filter` to have the same behavior, why not just extract most of the body, give it a return-type of `List<Func<QueryContainerDescriptor<T>, QueryContainer>>`, and use the return-value [on this line](https://github.com/ritterim/filter/pull/46/files#diff-3b4683a3f2a756be47c20cca315a352fR90)?